src/slave/containerizer/composing.cpp

```diff
@@ -425,6 +425,28 @@ Future<Containerizer::LaunchResult> ComposingContainerizerProcess::_launch(
 }
 
 
+static Try<Nothing> validateContainerInfoType(const ContainerInfo& info)
+{
+  const auto* typeDescriptor = ContainerInfo::Type_descriptor();
+  const auto* infoDescriptor = ContainerInfo::descriptor();
+  for (int index = 0; index < typeDescriptor->value_count(); index++) {
+    const auto* typeValueDescriptor = typeDescriptor->value(index);
+    const auto* infoValueDescriptor = infoDescriptor->FindFieldByName(
+      strings::lower(typeValueDescriptor->name()));
+    CHECK_NOTNULL(infoValueDescriptor);
+
+    if (
+      int(info.type()) != typeValueDescriptor->number() &&
```

Comment:
It looks like, in every protobuf union except this one, we have an `UNKNOWN = 0` Type, which guards against future additions to the protobuf.  If you end up generalizing this helper, the `0` case will need to be ignored.